### ISSUE TITLE
AKS ARM template, AKSLoggingEnabled.py, TypeError: string indices must be integers

### ISSUE THREAD

**Issue by aholler2** [bridgecrewio/checkov#1486]
**Describe the bug**
Python is evaluating the following input as true and therefore breaking on the next line. If I change the name of variable "omsagentAddonFinal" to say "dfasdfasdf", the test will run fine.
```python
if "omsagent" in conf["properties"]["addonProfiles"]:
```
```json
"addonProfiles": "[union(variables('omsagentAddonFinal'), variables('azurepolicyAddon'),variables('appgwIngressControllerAddonFinal'))]"
```

**To Reproduce**
```cmd
checkov --directory C:/Users/hollea/Documents/AzureTemplates/AzureArtifacts/templates/aks/ --output junitxml
2021-08-10 09:35:00,405 [MainThread  ] [ERROR]  Failed to run check: Ensure AKS logging to Azure Monitoring is Configured for configuration: {'dependsOn': ["[concat('Microsoft.Network/publicIPAddresses/', variables('publicIPAddressName'))]"], 'apiVersion': '2020-07-01', 'type': 'Microsoft.ContainerService/managedClusters', 'location': '[resourceGroup().location]', 'tags': {'__startline__': 179, '__endline__': 179}, 'name': "[parameters('clusterName')]", 'identity': {'type': 'SystemAssigned', '__startline__': 256, '__endline__': 258}, 'properties': {'kubernetesVersion': '', 'nodeResourceGroup': "[concat(resourceGroup().name, '-infra')]", 'dnsPrefix': "[parameters('clusterName')]", 'enableRBAC': True, 'networkProfile': {'networkPlugin': 'azure', 'networkPolicy': 'azure', 'outboundType': 'loadBalancer', 'loadBalancerSku': 'standard', 'loadBalancerProfile': {'outboundIPs': {'publicIPs': [{'id': "[resourceId('Microsoft.Network/publicIPAddresses',variables('publicIPAddressName'))]", '__startline__': 272, '__endline__': 274}], '__startline__': 270, '__endline__': 276}, '__startline__': 269, '__endline__': 277}, '__startline__': 264, '__endline__': 278}, 'agentPoolProfiles': [{'name': 'pool1', 'tags': "[parameters('agentPoolTags')]", 'type': 'VirtualMachineScaleSets', 'mode': 'System', 'osDiskSizeGB': 0, 'vnetSubnetID': "[if(empty(parameters('vnetSubnetID')), json('null'), parameters('vnetSubnetID'))]", 'count': 2, 'vmSize': 'Standard_B2s', 'osType': 'Linux', 'maxPods': 30, 'enableNodePublicIP': False, 'storageProfile': 'ManagedDisks', '__startline__': 280, '__endline__': 293}], 'aadProfile': {'managed': True, 'enableAzureRBAC': True, '__startline__': 295, '__endline__': 298}, 'apiServerAccessProfile': {'authorizedIPRanges': "[concat(variables('catIPs'),parameters('apiServerauthorizedIPRanges'))]", '__startline__': 299, '__endline__': 301}, 'addonProfiles': "[union(variables('omsagentAddonFinal'), variables('azurepolicyAddon'),variables('appgwIngressControllerAddonFinal'))]", '__startline__': 259, '__endline__': 303}, 'resources': [{'condition': "[not(empty(parameters('diagnosticSettings')))]", 'type': 'providers/diagnosticSettings', 'name': "[concat('/Microsoft.Insights/',variables('diagnosticLogssettingName'),'workspace')]", 'apiVersion': '2017-05-01-preview', 'properties': {'storageAccountId': "[if(contains(parameters('diagnosticSettings'),'storageAccountId'), parameters('diagnosticSettings').storageAccountId, json('null'))]", 'workspaceId': "[if(contains(parameters('diagnosticSettings'),'workspaceId'), parameters('diagnosticSettings').workspaceId, json('null'))]", 'eventHubAuthorizationRuleId': "[if(contains(parameters('diagnosticSettings'),'eventHubAuthorizationRuleId'), parameters('diagnosticSettings').eventHubAuthorizationRuleId, json('null'))]", 'eventHubName': "[if(contains(parameters('diagnosticSettings'),'eventHubName'), parameters('diagnosticSettings').eventHubName, json('null'))]", 'logs': [{'category': 'kube-apiserver', 'enabled': True, 'retentionPolicy': {'days': "[if(contains(parameters('diagnosticSettings'),'logRetentionDays'), parameters('diagnosticSettings').logRetentionDays, 90)]", 'enabled': True, '__startline__': 319, '__endline__': 322}, '__startline__': 316, '__endline__': 323}, {'category': 'kube-audit', 'enabled': False, 'retentionPolicy': {'days': "[if(contains(parameters('diagnosticSettings'),'logRetentionDays'), parameters('diagnosticSettings').logRetentionDays, 90)]", 'enabled': True, '__startline__': 327, '__endline__': 330}, '__startline__': 324, '__endline__': 331}, {'category': 'kube-controller-manager', 'enabled': True, 'retentionPolicy': {'days': "[if(contains(parameters('diagnosticSettings'),'logRetentionDays'), parameters('diagnosticSettings').logRetentionDays, 90)]", 'enabled': True, '__startline__': 335, '__endline__': 338}, '__startline__': 332, '__endline__': 339}, {'category': 'kube-scheduler', 'enabled': True, 'retentionPolicy': {'days': "[if(contains(parameters('diagnosticSettings'),'logRetentionDays'), parameters('diagnosticSettings').logRetentionDays, 90)]", 'enabled': True, '__startline__': 343, '__endline__': 346}, '__startline__': 340, '__endline__': 347}, {'category': 'cluster-autoscaler', 'enabled': True, 'retentionPolicy': {'days': "[if(contains(parameters('diagnosticSettings'),'logRetentionDays'), parameters('diagnosticSettings').logRetentionDays, 90)]", 'enabled': False, '__startline__': 351, '__endline__': 354}, '__startline__': 348, '__endline__': 355}], 'metrics': [{'enabled': True, 'category': 'AllMetrics', 'retentionPolicy': {'days': "[if(contains(parameters('diagnosticSettings'),'logRetentionDays'), parameters('diagnosticSettings').logRetentionDays, 90)]", 'enabled': False, '__startline__': 361, '__endline__': 364}, '__startline__': 358, '__endline__': 365}], '__startline__': 310, '__endline__': 367}, 'dependsOn': ["[concat('Microsoft.ContainerService/managedClusters/', parameters('clusterName'))]"], '__startline__': 305, '__endline__': 371, 'parent_name': "[parameters('clusterName')]", 'parent_type': 'Microsoft.ContainerService/managedClusters'}], '__startline__': 249, '__endline__': 373} at file: /aks-basic.json
Traceback (most recent call last):
  File "C:\Users\hollea\AppData\Local\Programs\Python\Python37\Scripts\checkov.cmd", line 53, in <module>
    run()
  File "C:\Users\hollea\AppData\Local\Programs\Python\Python37\lib\site-packages\checkov\main.py", line 167, in run
    files=file, guidelines=guidelines)
  File "C:\Users\hollea\AppData\Local\Programs\Python\Python37\lib\site-packages\checkov\common\runners\runner_registry.py", line 54, in run
    collect_skip_comments=collect_skip_comments,
  File "C:\Users\hollea\AppData\Local\Programs\Python\Python37\lib\site-packages\checkov\arm\runner.py", line 99, in run
    runner_filter)
  File "C:\Users\hollea\AppData\Local\Programs\Python\Python37\lib\site-packages\checkov\common\checks\base_check_registry.py", line 121, in scan
    result = self.run_check(check, entity_configuration, entity_name, entity_type, scanned_file, skip_info)
  File "C:\Users\hollea\AppData\Local\Programs\Python\Python37\lib\site-packages\checkov\common\checks\base_check_registry.py", line 140, in run_check
    skip_info=skip_info,
  File "C:\Users\hollea\AppData\Local\Programs\Python\Python37\lib\site-packages\checkov\common\checks\base_check.py", line 67, in run
    raise e
  File "C:\Users\hollea\AppData\Local\Programs\Python\Python37\lib\site-packages\checkov\common\checks\base_check.py", line 54, in run
    check_result["result"] = self.scan_entity_conf(entity_configuration, entity_type)
  File "C:\Users\hollea\AppData\Local\Programs\Python\Python37\lib\site-packages\checkov\arm\base_resource_check.py", line 16, in scan_entity_conf
    return self.scan_resource_conf(conf, entity_type)
  File "C:\Users\hollea\AppData\Local\Programs\Python\Python37\lib\site-packages\checkov\arm\base_resource_check.py", line 28, in wrapper
    return wrapped(self, conf)
  File "C:\Users\hollea\AppData\Local\Programs\Python\Python37\lib\site-packages\checkov\arm\checks\resource\AKSLoggingEnabled.py", line 22, in scan_resource_conf
    if "enabled" in conf["properties"]["addonProfiles"]["omsagent"]:
TypeError: string indices must be integers.
```

**Desktop (please complete the following information):**
 - OS: Windows
 - Checkov Version 2.0.344

**Additional context**
Full ARM Template:
```json
{
    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
    "contentVersion": "#{versionNumber}#",
    "metadata": {
      "title": "Azure Kubernetes Service",
      "description": "This template provisions an AKS Cluster"
    },
    "parameters": {
        "kubernetesVersion": {
            "type": "string",
            "metadata": {
                "description": "The version of the Managed Cluster resource, latest version used if no value is provided"
            },
            "defaultValue": ""
        },
        "clusterName": {
            "type": "string",
            "metadata": {
                "description": "(Required) The name of the Managed Cluster resource"
            }
        },
        "location": {
            "type": "string",
            "defaultValue": "[resourceGroup().location]",
            "metadata": {
                "description": "The location of the Managed Cluster resource"
            }
        },
        "omsWorkSpaceId": {
            "type": "string",
            "metadata": {
                "description": "(Required) Log analytics Workspace resource ID"
            }
        },
        "networkPlugin": {
            "type": "string",
            "metadata": {
                "description": "Network plugin used for building Kubernetes network - azure or kubenet"
            },
            "defaultValue": "azure",
            "allowedValues": [
                "azure",
                "kubenet"
            ]
        },
        "networkPolicy": {
            "type": "string",
            "metadata": {
                "description": "Network policy used for building Kubernetes network - calico or azure, azure network policy not allowed for kubenet networkPlugin"
            },
            "defaultValue": "azure",
            "allowedValues": [
                "azure",
                "calico"
            ]
        },
        "vnetSubnetID": {
            "type": "string",
            "metadata": {
                "description": "Existing Vnet Subnet resource id to be used to deploy the cluster worker nodes"
            },
            "defaultValue": ""
        },
        "nodeResourceGroup": {
            "type": "string",
            "metadata": {
                "description": "Name of the resource group containing agent pool nodes, will be generated if not provided"
            },
            "defaultValue": "[concat(resourceGroup().name, '-infra')]"
        },
        "agentPoolName": {
            "type": "string",
            "defaultValue": "pool1",
            "metadata": {
                "description": "The name of the agent pool"
            }
        },
        "agentPooltype": {
            "type": "string",
            "defaultValue": "VirtualMachineScaleSets",
            "allowedValues": [
                "VirtualMachineScaleSets",
                "AvailabilitySet"
            ],
            "metadata": {
                "description": "Type of Agent Pool VMs"
            }
        },
        "availabilityZones":{
            "type": "array",
            "defaultValue": [],
            "metadata": {
                "description": "Availability zones for nodes. Must use VirtualMachineScaleSets AgentPoolType"
            }
        },
        "agentCount": {
            "type": "int",
            "defaultValue": 2,
            "metadata": {
                "description": "The number of agent nodes for the cluster. Production workloads have a recommended minimum of 3"
            }
        },
        "agentVMSize": {
            "type": "string",
            "defaultValue": "Standard_B2s",
            "metadata": {
                "description": "The size of the Virtual Machine, for available sizes refer https://docs.microsoft.com/azure/templates/microsoft.containerservice/2020-04-01/managedclusters#managedclusteragentpoolprofile-object"
            }
        },       
        "osType": {
            "type": "string",
            "defaultValue": "Linux",
            "allowedValues": [
                "Linux"
            ],
            "metadata": {
                "description": "The type of operating system, kubenet networking model only supports Linux"
            }
        },
        "osDiskSizeGB": {
            "type": "int",
            "defaultValue": 0,
            "metadata": {
                "description": "Disk size (in GB) to provision for each of the agent pool nodes. This value ranges from 0 to 1023. Specifying 0 will apply the default disk size for that agentVMSize"
            },
            "minValue": 0,
            "maxValue": 1023
        },
        "maxPods": {
            "type": "int",
            "defaultValue": 30,
            "metadata": {
                "description": "Maximum number of pods that can run on a node"
            }
        },
        "applicationGatewayName": {
            "type": "string",
            "metadata": {
                "description": "AppGateway Name to be created - Standard Tier "
            },
            "defaultValue": "[concat(parameters('clusterName'), '-appgtwy')]"
        },
        "applicationGatewaySubnetPrefix": {
            "type": "string",
            "metadata": {
                "description": "Subnet Address Prefix in which the Application Gateway needs to be created"
            },
            "defaultValue": "10.2.0.0/16"
        },
        "diagnosticSettings": {
            "type": "object",
            "defaultValue": {},
            "metadata": {
                "description": "Enables Diagnostic Settings on the account. All log category will be enabled. Specify atleast one destination in the same Region as AKS(Storage Account/Log Analytics Workspace/Event Hub.",
                "comments": [
                    {
                        "workspaceId": "<(string) (Optional) Azure Resource ID of the Log Analytics workspace for the Log Analytics workspace to which logs will be sent.>",
                        "storageAccountId": "<(string) (Optional) Resource Id of the Storage Account in which platform logs should be saved.>",
                        "logRetentionDays": "<(int) Only applies to Storage Account, The number of days for the retention in days. A value of 0 will retain the events indefinitely.>",
                        "eventHubName": "<(string) Optional. Name of the event hub within the namespace to which logs are streamed. Without this, an event hub is created for each log category.>",
                        "eventHubAuthorizationRuleId": "<(string)Resource ID of the event hub authorization rule for the Event Hubs namespace in which the event hub should be created or streamed to.>"
                    }
                ]
            }
        },
        "kubeAuditLog":{
            "type": "bool",
            "metadata": {
                "description": "Kube Audit Log that logs every event in the Cluster, only enable for Troubleshooting to minimize Logging costs"
            },
            "defaultValue": false
        },
        "agentPooltags": {
            "type": "object",
            "defaultValue": {}
        },
        "clusterTags": {
            "type": "object",
            "defaultValue": {}
        },
        "apiServerauthorizedIPRanges": {
            "type": "array",
            "defaultValue": [],
            "metadata": {
                "description": "Authorized IP ranges for API Server"
            }
        }
    },
    "variables": {
        "IPs": [
            "1.1.1.1/32"
        ],
        "allIPs": "[concat(variables('IPs'),parameters('apiServerauthorizedIPRanges'))]",
        "emptyObject": {},
        "publicIPAddressName": "[concat(parameters('clusterName'),'-egress-pip')]",
        "publicIPRef": "[resourceId('Microsoft.Network/publicIPAddresses',variables('publicIPAddressName'))]",
        "diagnosticLogssettingName": "[concat(parameters('clusterName'),'Logs')]",
        "dnsPrefix": "[parameters('clusterName')]",
        "omsagentAddon": {
            "omsagent": {
                "enabled": true,
                "config": {
                    "logAnalyticsWorkspaceResourceID": "[parameters('omsWorkSpaceId')]"
                }
            }
        },
        "azurepolicyAddon": {
            "azurepolicy": {
                "enabled": true
            }
        },
        "appgwIngressControllerAddon": {
            "IngressApplicationGateway": {
                "enabled": true,
                "config": {
                    "applicationGatewayName": "[parameters('applicationGatewayName')]",
                    "subnetPrefix": "[parameters('applicationGatewaySubnetPrefix')]"
                }
            }
        },
        "omsagentAddonFinal": "[if(empty(parameters('omsWorkSpaceId')),variables('emptyObject'),variables('omsagentAddon'))]",
        "appgwIngressControllerAddonFinal": "[if(empty(parameters('applicationGatewayName')),variables('emptyObject'),variables('appgwIngressControllerAddon'))]",
        "addonsFinal": "[union(variables('omsagentAddonFinal'), variables('azurepolicyAddon'),variables('appgwIngressControllerAddonFinal'))]"
    },
    "resources": [
        {
            "name": "[variables('publicIPAddressName')]",
            "apiVersion": "2020-05-01",
            "type": "Microsoft.Network/publicIPAddresses",
            "location": "[parameters('location')]",
            "sku": {
                "name": "Standard"
            },
            "properties": {
                "publicIPAllocationMethod": "Static",
                "publicIPAddressVersion": "IPv4"
            }
        },
        {
            "dependsOn": ["[concat('Microsoft.Network/publicIPAddresses/', variables('publicIPAddressName'))]"],
            "apiVersion": "2020-07-01",
            "type": "Microsoft.ContainerService/managedClusters",
            "location": "[parameters('location')]",
            "tags": "[parameters('clusterTags')]",
            "name": "[parameters('clusterName')]",
            "identity": {
                "type": "SystemAssigned"
            },
            "properties": {
                "kubernetesVersion": "[parameters('kubernetesVersion')]",
                "nodeResourceGroup": "[parameters('nodeResourceGroup')]",
                "dnsPrefix": "[variables('dnsPrefix')]",
                "enableRBAC": true,
                "networkProfile": {
                    "networkPlugin": "[parameters('networkPlugin')]",
                    "networkPolicy": "[parameters('networkPolicy')]",
                    "outboundType": "loadBalancer",
                    "loadBalancerSku": "standard",
                    "loadBalancerProfile": {
                        "outboundIPs": {
                            "publicIPs": [
                              {
                                "id": "[variables('publicIPRef')]"
                              }
                            ]
                          }
                    }
                },
                "agentPoolProfiles": [
                    {
                        "name": "[parameters('agentPoolName')]",
                        "tags": "[parameters('agentPoolTags')]",
                        "type": "[parameters('agentPooltype')]",
                        "mode": "System",
                        "osDiskSizeGB": "[parameters('osDiskSizeGB')]",
                        "vnetSubnetID": "[if(empty(parameters('vnetSubnetID')), json('null'), parameters('vnetSubnetID'))]",
                        "count": "[parameters('agentCount')]",
                        "vmSize": "[parameters('agentVMSize')]",
                        "osType": "[parameters('osType')]",
                        "maxPods": "[parameters('maxPods')]",
                        "enableNodePublicIP": false,
                        "storageProfile": "ManagedDisks"
                    }
                ],
                "aadProfile": {
                    "managed": true,
                    "enableAzureRBAC": true
                },
                "apiServerAccessProfile": {
                    "authorizedIPRanges": "[variables('allIPs')]"
                },
                "addonProfiles": "[variables('addonsFinal')]"
            },
            "resources": [
                {
                    "condition": "[not(empty(parameters('diagnosticSettings')))]",
                    "type": "providers/diagnosticSettings",
                    "name": "[concat('/Microsoft.Insights/',variables('diagnosticLogssettingName'),'workspace')]",
                    "apiVersion": "2017-05-01-preview",
                    "properties": {
                        "storageAccountId": "[if(contains(parameters('diagnosticSettings'),'storageAccountId'), parameters('diagnosticSettings').storageAccountId, json('null'))]",
                        "workspaceId": "[if(contains(parameters('diagnosticSettings'),'workspaceId'), parameters('diagnosticSettings').workspaceId, json('null'))]",
                        "eventHubAuthorizationRuleId": "[if(contains(parameters('diagnosticSettings'),'eventHubAuthorizationRuleId'), parameters('diagnosticSettings').eventHubAuthorizationRuleId, json('null'))]",
                        "eventHubName": "[if(contains(parameters('diagnosticSettings'),'eventHubName'), parameters('diagnosticSettings').eventHubName, json('null'))]",
                        "logs": [
                            {
                                "category": "kube-apiserver",
                                "enabled": true,
                                "retentionPolicy": {
                                    "days": "[if(contains(parameters('diagnosticSettings'),'logRetentionDays'), parameters('diagnosticSettings').logRetentionDays, 90)]",
                                    "enabled": true
                                }
                            },
                            {
                                "category": "kube-audit",
                                "enabled": "[parameters('kubeAuditLog')]",
                                "retentionPolicy": {
                                    "days": "[if(contains(parameters('diagnosticSettings'),'logRetentionDays'), parameters('diagnosticSettings').logRetentionDays, 90)]",
                                    "enabled": true
                                }
                            },
                            {
                                "category": "kube-controller-manager",
                                "enabled": true,
                                "retentionPolicy": {
                                    "days": "[if(contains(parameters('diagnosticSettings'),'logRetentionDays'), parameters('diagnosticSettings').logRetentionDays, 90)]",
                                    "enabled": true
                                }
                            },
                            {
                                "category": "kube-scheduler",
                                "enabled": true,
                                "retentionPolicy": {
                                    "days": "[if(contains(parameters('diagnosticSettings'),'logRetentionDays'), parameters('diagnosticSettings').logRetentionDays, 90)]",
                                    "enabled": true
                                }
                            },
                            {
                                "category": "cluster-autoscaler",
                                "enabled": true,
                                "retentionPolicy": {
                                    "days": "[if(contains(parameters('diagnosticSettings'),'logRetentionDays'), parameters('diagnosticSettings').logRetentionDays, 90)]",
                                    "enabled": false
                                }
                            }
                        ],
                        "metrics": [
                            {
                                "enabled": true,
                                "category": "AllMetrics",
                                "retentionPolicy": {
                                    "days": "[if(contains(parameters('diagnosticSettings'),'logRetentionDays'), parameters('diagnosticSettings').logRetentionDays, 90)]",
                                    "enabled": false
                                }
                            }
                        ]
                    },
                    "dependsOn": [
                        "[concat('Microsoft.ContainerService/managedClusters/', parameters('clusterName'))]"
                    ]
                }
            ]
        }
    ],
    "outputs": {
        "aksControlPlaneFQDN": {
            "type": "string",
            "value": "[reference(parameters('clusterName')).fqdn]"
        },
        "aksResourceId": {
            "type": "string",
            "value": "[resourceId('Microsoft.ContainerService/managedClusters',parameters('clusterName'))]"
        },
        "clusterName": {
            "type": "string",
            "value": "[parameters('clusterName')]"
        },
        "aksManagedIdentity": {
            "type": "string",
            "value": "[reference(parameters('clusterName'), '2020-07-01', 'Full').identity.principalId]"
        },
        "agentPoolManagedIdentityName": {
            "type": "string",
            "value": "[concat(parameters('clusterName'), '-agentpool')]"
        },
        "nodeResourceGroupName": {
            "type": "string",
            "value": "[parameters('nodeResourceGroup')]"
        },
        "aksAppGatewayName": {
            "type": "string",
            "value": "[parameters('applicationGatewayName')]"
        },
        "aksAppGatewaySubnetAddressPrefix": {
            "type": "string",
            "value": "[parameters('applicationGatewaySubnetPrefix')]"
        }
    }
}
```

**Comment by aholler2** [comment]
Resolved by #1763. Thank you @Arun-kc!

**Comment by Arun-kc** [comment]
You are welcome @aholler2
